subprojects/core/src/main/java/org/gradle/api/internal/initialization/ScriptHandlerInternal.java

```diff
@@ -20,9 +20,16 @@
 import org.gradle.internal.classpath.ClassPath;
 
 public interface ScriptHandlerInternal extends ScriptHandler {
+
     void addScriptClassPathDependency(Object notation);
 
+    /**
+     * @return The script classpath as used at runtime.
+     */
     ClassPath getScriptClassPath();
 
-
+    /**
+     * @return The script classpath as declared.
+     */
+    ClassPath getDeclaredScriptClassPath();
```

Comment:
Nit: this isn't really "as declared" as a user would only declare the first level plugins/dependencies. It's more the "unprocessed" or "uninstrumented" classpath.